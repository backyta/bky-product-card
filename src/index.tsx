
export * from './components';
























//* Aqui se ponen todas las exportacion que van a ser vistas desde el mundo exterior, usualmente solo
//* se exponen los componentes es lo que exponemos.