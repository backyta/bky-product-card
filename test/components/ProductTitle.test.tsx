import React from 'react'
import renderer from 'react-test-renderer'
import { ProductTitle, ProductCard } from '../../src/components' 
import { product } from '../data/products';



describe('Pruebas en ProductTitle', () => {

    test('debe de mostrar el componente correctamente con el titulo perzonalizado', () => {
        
        const wrapper = renderer.create(
            <ProductTitle title='Custom Product' className='Custom-class'/>
        )
        // console.log( wrapper.toJSON() );
        expect( wrapper.toJSON() ).toMatchSnapshot();


    });
    
    test('debe de mostrar el componete con el nombre del producto', () => {

        const wrapper = renderer.create(
            <ProductCard product={ product }>
                {
                    () =>(
                        <ProductTitle />
                    )
                }
            </ProductCard>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
        //* Si no manda el title en las props del ProductTitle lo toma del context del product card
        //* del producto.
    });
});