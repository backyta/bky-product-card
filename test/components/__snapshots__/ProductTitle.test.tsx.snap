// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductTitle debe de mostrar el componente correctamente con el titulo perzonalizado 1`] = `
<span
  className="productDescription Custom-class"
>
  Custom Product
</span>
`;

exports[`Pruebas en ProductTitle debe de mostrar el componete con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Sin imagen
  </span>
</div>
`;
